selftests/functional/test_output.py

```diff
@@ -391,11 +379,11 @@ def test_redirect_output(self):
     @unittest.skipIf(perl_tap_parser_uncapable(),
                      "Uncapable of using Perl TAP::Parser library")
     def test_tap_parser(self):
-        perl_script = script.TemporaryScript("tap_parser.pl",
-                                             PERL_TAP_PARSER_SNIPPET)
-        perl_script.save()
         os.chdir(basedir)
-        process.run("perl %s" % perl_script)
+        result = process.run(os.path.join("selftests", ".scripts",
```

Comment:
IIRC we talked about it and Ademar suggested in some cases it's probably worth doing so and I should send PR to fix the most pressing ones.

Basically simple snippets should be used as scripts, but complex examples which is useful to have to execute manually could be extracted. And in this example I think it's worth doing that, because imagine you want to debug that code as the script does not provide much info.